native/src/main/java/io/ballerina/observe/trace/zipkin/ZipkinTracerProvider.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.observe.trace.zipkin;
+
+import io.ballerina.observe.trace.zipkin.sampler.RateLimitingSampler;
+import io.ballerina.runtime.api.values.BDecimal;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.observability.tracer.spi.TracerProvider;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.exporter.zipkin.ZipkinSpanExporter;
+import io.opentelemetry.extension.trace.propagation.B3Propagator;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.SdkTracerProviderBuilder;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+
+import java.io.PrintStream;
+import java.util.concurrent.TimeUnit;
+
+import static io.opentelemetry.semconv.resource.attributes.ResourceAttributes.SERVICE_NAME;
+
+/**
+ * This is the Zipkin tracing extension class for {@link TracerProvider}.
+ */
+public class ZipkinTracerProvider implements TracerProvider {
+    private static final String TRACER_NAME = "zipkin";
+    private static final PrintStream console = System.out;
+
+    static SdkTracerProviderBuilder tracerProviderBuilder;
+
+    @Override
+    public String getName() {
+        return TRACER_NAME;
+    }
+
+    @Override
+    public void init() {    // Do Nothing
+    }
+
+    public static void initializeConfigurations(BString agentHostname, int agentPort, BString samplerType,
+                                                BDecimal samplerParam, int reporterFlushInterval,
+                                                int reporterBufferSize) {
+
+        String reporterEndpoint = "http://" + agentHostname + ":" + agentPort + "/api/v2/spans";
```

Comment:
@NipunaMadhushan Shall we make `http` configurable? Also can we make `https` by default?

Reply:
Addressed the suggestions